card/lib/card/set/event.rb

```diff
@@ -96,10 +96,10 @@ def event_type
       end
 
       def define_standard_event_method method_name=simple_method_name
-        stage = @stage
+        is_integration = @stage.to_s.match?(/integrate/)
         @set_module.class_exec(@event) do |event_name|
           define_method event_name do
-            rescuing_integration stage do
+            rescuing_if_integration is_integration do
```

Comment:
I was seeing some errors saying "stage not defined", but I didn't really have a good way to reproduce them.  So I'm not sure this will help, but it seems a little more efficient anyway...